crux/src/Crux.hs

```diff
@@ -648,6 +648,10 @@ doSimWithResults cruxOpts simCallback sym execFeatures profInfo monline goalProv
       let ctx = execResultContext res
       inFrame profInfo "<Prove Goals>" $ do
         todo <- getProofObligations sym
+        let showGoal = \ g -> ( T.pack (show (view labeledPred g))
+                              , T.pack (show (view labeledPredMsg g))
+                              )
```

Comment:
Would it be better to just log the goal object itself here and then let the logger decide which portions it would like to show?

Reply:
Reasonable question. I had two concerns:

1. JSON serializability, but since we had talked about putting `Doc`s in the log data types, I think it no longer makes sense to try and have the `...LogMessage` data types have derivable JSON instances (as I originally did), and so storing more complex data in the log messages would no longer be a concern accordingly.

2. Type indices, those goals are at the very least indexed by `sym`, so we would have to choose whether to existentially wrap it in the `Goals` constructor, or reflect it in the type of `CruxLogMessage`. With the former, we no longer get deriving for the `Generic` instance, so we must write all `ToJSON` cases by hand without `genericToEncoding`. With the latter, I'm not sure whether it creates problems, it's just a little more noise but maybe it's all fine?

I'm attempting this change locally, will report on the outcome.

Reply:
OK.  The second concern seems fairly significant, so I wouldn't go to extraordinary lengths to make this work.  In general, my thoughts are that whenever reasonable, we can defer the output representation to the logger instead of the point of log, but it will be nice to keep these relatively simple as well rather than incurring  a significant technical burden just for logging/output.

Reply:
I have attempted the existential wrapping, and it's not too bad: d91b9978b83feb6e9ba8b97b5419d4080e7d0ec7

We need to have `sym` be partially instantiated to `ExprBuilder t st fs` so as to have the `Show` instance without adding constraints. I'm happy with this change.